py-polars/polars/io.py

```diff
@@ -209,7 +209,7 @@ def read_csv(
 
     if columns and not has_header:
         for column in columns:
-            if isinstance(column, str) and not column.startswith("column_"):
+            if not column.startswith("column_"):
```

Comment:
`columns` is, before this statement, latest touched by `handle_projection_columns`, and can be a list of ints. Thus we cannot assume that `column` is always a string?

Reply:
If you check the `handle_projection_columns` function, it makes sure that `columns` is either None or a list of strings. That's why the `redundant-expr` error triggered!

Reply:
Yes, you are right, I misread the type annotation `tuple[list[int] | None, list[str] | None]`, the first is for `projection`, not `columns`.